joern-cli/frontends/gosrc2cpg/src/test/scala/io/joern/go2cpg/passes/ast/AstCreationForTypeDeclTests.scala

```diff
@@ -0,0 +1,256 @@
+
+    "test the modifier" in {
+      typeDeclNode.modifier.head.modifierType shouldBe ModifierTypes.PUBLIC
+      typeDeclNode.astOut.isModifier.l.size shouldBe 1
+    }
+
+  }
+
+  "when a single TypeDecl is defined using a prefix" should {
+    val cpg = code(
+      """package main
+        |type (
+        | Foo struct {}
+        |)""".stripMargin,
+      "test.go"
+    )
+
+    val List(typeDeclNode) = cpg.typeDecl.nameExact("Foo").l
+    "test basic ast structure" in {
+      typeDeclNode.code shouldBe "Foo struct {}"
+      typeDeclNode.lineNumber shouldBe Some(3)
+      typeDeclNode.columnNumber shouldBe Some(2)
+    }
+
+    "test ast parent structure" in {
+      cpg.typeDecl("Foo").astParentFullName.l.head shouldBe "test.go:main.<global>"
+      cpg.typeDecl("Foo").astParentType.l.head shouldBe NodeTypes.TYPE_DECL
+    }
+
+    "test fullName of TypeDecl nodes" ignore {
+      typeDeclNode.fullName shouldBe "test.go::main.<global>::main::Foo"
+    }
+
+    "test the modifier" in {
+      typeDeclNode.modifier.head.modifierType shouldBe ModifierTypes.PUBLIC
+      typeDeclNode.astOut.isModifier.l.size shouldBe 1
+    }
+
+  }
+
+  "when multiple TypeDecls are defined using a prefix" should {
+    val cpg = code(
+      """package main
+        |type (
+        | Foo struct {}
+        | bar interface {}
+        |)""".stripMargin,
+      "test.go"
+    )
+    val List(typeDeclNode) = cpg.typeDecl.nameExact("Foo").l
+    "test basic ast structure for Foo" in {
+      typeDeclNode.code shouldBe "Foo struct {}"
+      typeDeclNode.lineNumber shouldBe Some(3)
+      typeDeclNode.columnNumber shouldBe Some(2)
+    }
+
+    "test ast parent structure for Foo" in {
+      cpg.typeDecl("Foo").astParentFullName.l.head shouldBe "test.go:main.<global>"
+      cpg.typeDecl("Foo").astParentType.l.head shouldBe NodeTypes.TYPE_DECL
+    }
+
+    "test fullName of TypeDecl nodes for Foo" ignore {
+      typeDeclNode.fullName shouldBe "test.go::main.<global>::main::Foo"
+    }
+
+    "test the modifier for Foo" in {
+      typeDeclNode.modifier.head.modifierType shouldBe ModifierTypes.PUBLIC
+      typeDeclNode.astOut.isModifier.l.size shouldBe 1
+    }
+
+    val List(typeDeclNodeBar) = cpg.typeDecl.nameExact("bar").l
+    "test basic ast structure for bar" in {
+      typeDeclNodeBar.code shouldBe "bar interface {}"
+      typeDeclNodeBar.lineNumber shouldBe Some(4)
+      typeDeclNodeBar.columnNumber shouldBe Some(2)
+    }
+
+    "test parent ast structure for bar" in {
+      cpg.typeDecl("bar").astParentFullName.l.head shouldBe "test.go:main.<global>"
+      cpg.typeDecl("bar").astParentType.l.head shouldBe NodeTypes.TYPE_DECL
+    }
+
+    "test fullName of TypeDecl nodes for bar" ignore {
+      typeDeclNodeBar.fullName shouldBe "test.go::main.<global>::bar"
+    }
+
+    "test the modifier for bar" in {
+      typeDeclNodeBar.modifier.head.modifierType shouldBe ModifierTypes.PRIVATE
+      typeDeclNodeBar.astOut.isModifier.l.size shouldBe 1
```

Comment:
Going to list is usually unnecessary, same for the above
```suggestion
      typeDeclNodeBar.astOut.isModifier.size shouldBe 1
```